### PR TITLE
etcd-tester: skip compaction after different hash

### PR DESCRIPTION
**Pull request by gyuho**
When hashes don't match, there could be some nodes
falling behind and the compact request can then error
with 'future revision compact'.

### CODE REVIEW

**Comment by heyitsanthony** [comment]
Why not quorum/linearized Get before compacting to ensure all members caught up?

File: tools/functional-tester/etcd-tester/tester.go [hunk 1]
```diff
@@ -137,6 +141,9 @@ func (tt *tester) runLoop() {
 			plog.Printf("[round#%d case#%d] succeed!", i, j)
 		}
 
+		if differentHash {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: if !succeeded {
    continue
}

File: tools/functional-tester/etcd-tester/tester.go [hunk 2]
```diff
@@ -137,6 +141,9 @@ func (tt *tester) runLoop() {
 			plog.Printf("[round#%d case#%d] succeed!", i, j)
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: succeed = true

**Comment by xiang90** [comment]
@heyitsanthony I think the problem @gyuho is trying to solve here is to not continue to do compaction once the test already failed.

**Comment by gyuho** [comment]
@xiang90 PTAL.

@heyitsanthony Yeah the root cause was the failure before the compaction error. This commit doesn't solve the problem actually. This is more like to produce cleaner logs for debugging so that we don't get duplicate failure logs.

Thanks!

**Comment by xiang90** [comment]
@gyuho CI failed?

**Comment by gyuho** [comment]
@xiang90 I don't think it's related?

1. https://travis-ci.org/coreos/etcd/jobs/124624880#L1029
2. https://semaphoreci.com/coreos/etcd/branches/pull-request-5145/builds/4 - https://github.com/coreos/etcd/issues/5147 created an issue

**Comment by gyuho** [comment]
@xiang90 Ping?

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
thanks. will redeploy after I finish up the release.

**Pull request merged**